tests/utils.py

```diff
@@ -0,0 +1,273 @@
+    model_metadata = parse_trustyai_model_metadata(model_metadata=response.content)
+
+    assert (
+        model_metadata.model_name == model.name
+    ), f"Expected model name '{model.name}', but got '{model_metadata.model_name}'"
+    assert (
+        model_metadata.input_tensor_name == model_input_data.name
+    ), f"Expected input tensor name '{model_input_data}', but got '{model_metadata.input_tensor_name}'"
+
+    assert (
+        model_metadata.num_observations
+        > model_input_data.num_observations * expected_percentage_observations
+    ), (
+        f"Expected number of observations to be greater than "
+        f"{model_input_data.num_observations * expected_percentage_observations},"
+        f" but got {model_metadata.num_observations}"
+    )
+    assert (
+        model_metadata.num_features == model_input_data.num_features
+    ), f"Expected number of features '{model_input_data.num_features}', but got '{model_metadata.num_features}'"
+
+
+def parse_trustyai_model_metadata(model_metadata):
+    json_data = json.loads(model_metadata)
+
+    if isinstance(json_data, list) and len(json_data) > 0:
+        data = json_data[0]["data"]
+    else:
+        raise ValueError("Invalid JSON data format")
+
+    input_tensor_name = data["inputTensorName"]
+    output_tensor_name = data["outputTensorName"]
+    num_observations = data["observations"]
+    model_name = data["modelId"]
+
+    model_metadata = TrustyAIModelMetadata(
+        input_tensor_name=input_tensor_name,
+        output_tensor_name=output_tensor_name,
+        num_observations=num_observations,
+        model_name=model_name,
+        num_features=len(data["inputSchema"]["items"]),
+    )
+    return model_metadata
+
+
+def parse_input_data(data_path):
+    total_observations = 0
+    name = ""
+    num_features = 0
+    data_type = ""
+
+    for filename in os.listdir(data_path):
+        if filename.endswith(".json"):
+            file_path = os.path.join(data_path, filename)
+
+            with open(file_path, "r") as file:
+                data = json.load(file)
+
+            for input_data in data["inputs"]:
+                if not name:
+                    name = input_data["name"]
+                if not num_features:
+                    num_features = input_data["shape"][1]
+                if not data_type:
+                    data_type = input_data["datatype"]
+
+                num_observations = input_data["shape"][0]
+                total_observations += num_observations
+
+    model_input_data = ModelInputData(
+        name=name,
+        num_features=num_features,
+        num_observations=total_observations,
+        data_type=data_type,
+    )
+    return model_input_data
+
+
+def wait_for_model_pods_registered(client, namespace):
+    """Wait for model pods to be registered by TrustyAIService"""
+    pods_with_env_var = False
+    all_pods_running = False
+    timeout = 60 * 3
+    start_time = time()
+    while (
```

Comment:
can `TimeOutSampler` be used here? 

https://github.com/RedHatQE/timeout-sampler

Reply:
I think so, added a TODO to consider using TimeoutSampler if the framework grows